### DISCUSSION TITLE
how to configure using environment variables

### DISCUSSION THREAD

**Discussion by batusai513** [pnpm/pnpm#6566] [Q&A]
Hello, I'm trying to figure out how to configure pnpm using environment variables, I've checked the [documentation](https://pnpm.io/npmrc) and it seems possible:
> pnpm gets its configuration from the command line, environment variables, and .npmrc files

but it is not very clear how to achieve this, I suppose that as in `npm`, all configs will be prefixed, so for `store-dir` will become `pnpn_config_store_dir`, but that did not seems to work, what is the correct format to configure pnpm using env variables and what configs can be change using this method?

what I want to achieve is to configure pnpm to not move anything that writes into disk to `/tmp/.pnpm` folder
```
store-dir=/tmp/.pnpm/store
modules-dir=/tmp/.pnpm/node_modules
global-dir=/tmp/.pnpm/global
global-bin-dir=/tmp/.pnpm
state-dir=/tmp/.pnpm
cache-dir=/tmp/.pnpm/cache
```

I also tried adding all of those as command line arguments, but discovered that `global-bin-dir` and `state-dir` are treated as unknown arguments by the cli

Thanks.

**Comment by AstroGD** [comment]
Hey there,
Found your issue and figured out how it works, so I thought I'll share:

You have to prefix the config vars with `npm_config_` and replace all `-` with `_`.
So the `store-dir` option becomes the `npm_config_store_dir` environment variable.

Hope this helps ^^

**Comment by unikitty37** [comment]
For those that have had difficulty working out how to set command-line options (or even the prefix mentioned above), it appears the pnpm documentation is based on the npm documentation, except the two paragraphs that tell you how to do it are missing from the pnpm version.

https://docs.npmjs.com/cli/v11/using-npm/config#command-line-flags has the information, and could probably do with being added to the pnpm documentation as well 🙂

**Reply by equinusocio** [reply]
This has been [asked 3 years ago](https://github.com/orgs/pnpm/discussions/5727), they have no plans to add support for env vars, at least not intentionally. They still wann us to rely on the deprecated cross-env.
